azure_functions_worker/dispatcher.py

```diff
@@ -518,10 +528,21 @@ def gen(resp_queue):
 class AsyncLoggingHandler(logging.Handler):
 
     def emit(self, record: LogRecord) -> None:
-        # Since we disable console log after gRPC channel is initiated
-        # We should redirect all the messages into dispatcher
+        # Since we disable console log after gRPC channel is initiated,
+        # we should redirect all the messages into dispatcher.
+
+        # When dispatcher receives an exception, it should switch back
+        # to console logging. However, it is possible that
+        # __current_dispatcher__ is set to None as there are still messages
+        # buffered in this handler, not calling the emit yet.
         msg = self.format(record)
-        Dispatcher.current.on_logging(record, msg)
+        try:
+            Dispatcher.current.on_logging(record, msg)
+        except RuntimeError:
+            # This will cause 'Dispatcher not found' failure.
+            # Logging such of an issue will cause infinite loop of gRPC logging
+            # To mitigate, we should suppress the 2nd level error logging here.
+            pass
```

Comment:
Can we add a telemetry event - something which we can see in Kusto, without logging? For my info.